File: src/BaseHmiTypes/Screens/Shapes/HmiGraphicView.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Shapes;

public class HmiGraphicView : HmiSurfaceShapeBase
{
    public HmiGraphicView()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiGraphicView;
    }

    //TODO: we have here Source and Image. Source used in Unified, Image in Adavnced and WinnCC (as blob uri atm)
    public HmiProperty<string>? Source { get; set; }

    public HmiProperty<HmiImageSource>? Image { get; set; }

    public HmiProperty<HmiImageSource>? AlternateImage { get; set; }

    public HmiProperty<int>? GraphicStretchMode { get; set; }
}

File: typescript/base-hmi-types/src/screens/shapes/HmiGraphicView.ts
import { HmiColor, HmiHorizontalAlignment, HmiImageSource, HmiProperty, HmiVerticalAlignment } from "../base.js";
import { HmiSurfaceShapeBase } from "./HmiSurfaceShapeBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";

export class HmiGraphicView extends HmiSurfaceShapeBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiGraphicView;
  }

  source?: HmiProperty<string>;
  image?: HmiProperty<HmiImageSource>;
  alternateImage?: HmiProperty<HmiImageSource>;
  imageScaled?: HmiProperty<boolean>;
  imageBlink?: HmiProperty<boolean>;
  imageColor?: HmiProperty<HmiColor>;
  imageBackgroundColor?: HmiProperty<HmiColor>;
  imageBackgroundTransparent?: HmiProperty<boolean>;
  imageHorizontalAlignment?: HmiProperty<HmiHorizontalAlignment>;
  imageVerticalAlignment?: HmiProperty<HmiVerticalAlignment>;
  graphicStretchMode?: HmiProperty<number>;
}
